server/routerlicious/packages/lambdas/src/nexus/connect.ts

```diff
@@ -499,52 +499,56 @@ async function addMessageClientToClientManager(
 	}
 }
 
+/**
+ * Sets up a listener for broadcast signals from external API and broadcasts them to the room.
+ *
+ * @param socket - Websocket instance
+ * @param room - Current room
+ * @param lambdaDependencies - Lambda dependencies including collaborationSessionEventEmitter and logger
+ * @returns Dispose function to remove the added listener
+ */
 function setUpSignalListenerForRoomBroadcasting(
 	socket: IWebSocket,
 	room: IRoom,
-	documentId: string,
-	tenantId: string,
 	{ collaborationSessionEventEmitter, logger }: INexusLambdaDependencies,
-): void {
-	collaborationSessionEventEmitter?.on(
-		"broadcastSignal",
-		// eslint-disable-next-line @typescript-eslint/no-misused-promises
-		async (broadcastSignal: IBroadcastSignalEventPayload) => {
-			const { signalRoom, signalContent } = broadcastSignal;
-
-			// No-op if the room (collab session) that signal came in from is different
-			// than the current room. We reuse websockets so there could be multiple rooms
-			// that we are sending the signal to, and we don't want to do that.
-			if (
-				signalRoom.documentId === room.documentId &&
-				signalRoom.tenantId === room.tenantId
-			) {
+): () => void {
+	const broadCastSignalListener = (broadcastSignal: IBroadcastSignalEventPayload): void => {
+		const { signalRoom, signalContent } = broadcastSignal;
+
+		// No-op if the room (collab session) that signal came in from is different
+		// than the current room. We reuse websockets so there could be multiple rooms
+		// that we are sending the signal to, and we don't want to do that.
+		if (signalRoom.documentId === room.documentId && signalRoom.tenantId === room.tenantId) {
+			try {
+				const runtimeMessage = createRuntimeMessage(signalContent);
+
 				try {
-					const runtimeMessage = createRuntimeMessage(signalContent);
-
-					try {
-						socket.emitToRoom(getRoomId(signalRoom), "signal", runtimeMessage);
-					} catch (error) {
-						const errorMsg = `Failed to broadcast signal from external API.`;
-						Lumberjack.error(
-							errorMsg,
-							getLumberBaseProperties(signalRoom.documentId, signalRoom.tenantId),
-							error,
-						);
-					}
+					socket.emitToRoom(getRoomId(signalRoom), "signal", runtimeMessage);
```

Comment:
Just for my understanding: this is the place where the listener is holding the socket object. And it is implicitly holder the socket object because it is part of the input arguments. And we do this for every connected socket. Is it correct?

Reply:
I guess we have listeners in other places as well. Should we call the off for these listeners as well?

Reply:
No, this is just an `emit` call. The problem place was the `collaborationSessionEventEmitter.on`